File: src/main/java/com/bedirhan/cityeventmonitor/model/News.java
package com.bedirhan.cityeventmonitor.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Data
@Document(collection = "news")
public class News {

    @Id
    private String id;

    private String title;
    private String content;
    private NewsType type;

    private String locationText;

    private String district;

    private double latitude;
    private double longitude;

    private boolean geocodingFailed;

    private Set<String> sources = new HashSet<>();
    private Set<String> urls = new HashSet<>();

    private LocalDateTime publishDate;
}

File: frontend/src/types/news.ts
export type NewsType =
  | 'TRAFIK_KAZASI'
  | 'YANGIN'
  | 'ELEKTRIK_KESINTISI'
  | 'HIRSIZLIK'
  | 'KULTUREL_ETKINLIK'
  | 'DIGER';

export interface News {
  id: string;
  title: string;
  content: string;
  type: NewsType | null;
  district: string | null;
  locationText: string | null;
  latitude: number;
  longitude: number;
  geocodingFailed: boolean;
  sources: string[];
  urls: string[];
  publishDate: string; // ISO date string
}

